code-challenges.test.js

```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
+const secretCodeWord4 = "EeEe"
+const secretCodeWord5 = "Supercalifragilisticexpialidocious"
+
+// input: a string
+
+// output: a string with the following input letters changed to numbers. "a" to 4, "e" to 3, "i" to 1, and "o" to 0
+
+// pseudo:
+// Create a function called superSecretCodingMachine that takes a string as a parameter. Use .replaceAll() to replace all instances of the searched letter with the regex search pattern. Use /g to search all occurences of the letter in the string and /i to be case insensitive, then use a comma to separate the search pattern from the replacement value and add corresponding replacement value. Repeat this .replaceALL pattern three additional times, changing to the required corresponding searched letter and its number replacement. Return in one line off the single passed through string parameter.
+
+// test:
+// describe("superSecretCodingMachine", () => {
+//     it("take in a string and returns a string with a coded message" , () => {
+//         const secretCodeWord1 = "Lackadaisical"
+//         const secretCodeWord2 = "Gobbledygook"
+//         const secretCodeWord3 = "Eccentric"
```

Comment:
GREAT hoisting! This makes tests look so much cleaner and it's efficient as well.